File: nysol_python/mcmd/methods/convTex/db/smp/func_regexlen.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='regexlen'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id,str
1,xcbbbayy
2,xxcbaay
3,
4,bacabbca
'''
db['idatas'].append(data)

data={}
data['name']='dat2.csv'
data['text']='''
id,str
1,漢漢あbbbいyy
2,漢あbいいy
3,
4,bあいあbbいあ
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
正規表現 ``c.*a`` に最も長くマッチする部分文字列の長さを得る。
マッチした部分文字列については ``regexstr`` と同じ入力データを使っているので、比較すると分かりやすい。
'''
script['sh_code']='''
mcal c='regexlen($s{str},"c.*a")' a=rsl i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcal(c='regexlen($s{str},"c.*a")', a='rsl', i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='マルチバイト文字'
script['comment']='''
正規表現 ``"あ.*い"`` に最も長くマッチする部分文字列の長さを得る。
ただし、以下ではマルチバイト文字対応でないregexlen関数を使っているので、
文字数ではなくバイト数を返している。
'''
script['sh_code']='''
mcal c='regexlen($s{str},"あ.*い")' a=rsl i=dat2.csv o=rsl2.csv
'''
script['py_code']='''
nm.mcal(c='regexlen($s{str},"あ.*い")', a='rsl', i="dat2.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='マルチバイト文字2'
script['comment']='''
正規表現 ``"あ.*い"`` に最も長くマッチする部分文字列の長さを得る。
regexlenw関数を使うと、正しくマルチバイト文字を扱ってくれる。
'''
script['sh_code']='''
mcal c='regexlenw($s{str},"あ.*い")' a=rsl i=dat2.csv o=rsl3.csv
'''
script['py_code']='''
nm.mcal(c='regexlenw($s{str},"あ.*い")', a='rsl', i="dat2.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/func_regexlen.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id,str
1,xcbbbayy
2,xxcbaay
3,
4,bacabbca
EOF
)}

File.open("dat2.csv","w"){|fpw| fpw.write(
<<'EOF'
id,str
1,漢漢あbbbいyy
2,漢あbいいy
3,
4,bあいあbbいあ
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Find out the length of the longest substring that matches with the regular expression \verb|c.*a|. Since the same input data is used for matching substring in the \verb|regexstr| function, it is easier to compare the results.
EOF
scp=<<'EOF'
more dat1.csv
mcal c='regexlen($s{str},"c.*a")' a=rsl i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Multibyte characters"
comment=<<'EOF'
Find out the length of the longest substring that matches \verb|"い.*あ"|. However, since regexlen function do not support multibyte character, it returns the number of bytes instead of number of characters.
EOF
scp=<<'EOF'
more dat2.csv
mcal c='regexlen($s{str},"あ.*い")' a=rsl i=dat2.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Multibyte characters 2"
comment=<<'EOF'
Find out the length of the longest substring that matches \verb|"い.*あ"|. The \verb|regexlenw| function is able to process multibyte characters to count the number of characters.
EOF
scp=<<'EOF'
mcal c='regexlenw($s{str},"あ.*い")' a=rsl i=dat2.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)

